src/main/java/digital/slovensko/autogram/core/eforms/EFormUtils.java

```diff
@@ -0,0 +1,284 @@
+
+        var builderFactory = DocumentBuilderFactory.newInstance();
+        builderFactory.setNamespaceAware(true);
+
+        Document responseDocument = null;
+        try {
+            builderFactory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+            responseDocument = builderFactory.newDocumentBuilder().newDocument();
+        } catch (ParserConfigurationException e) {
+            throw new UnrecognizedException(e);
+        }
+
+        if (xmlData.getFirstChild() == null)
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Element XMLData je prázdny");
+
+        var idk = xmlData.getFirstChild();
+        if (idk.getNodeType() == Node.TEXT_NODE) {
+            if (!idk.getNodeValue().matches("\\s*"))
+                throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "XMLData obsahuje neplatný text");
+
+            idk = idk.getNextSibling();
+        }
+
+        if (idk == null)
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Element XMLData je prázdny");
+
+        try {
+            var node = responseDocument.importNode(idk, true);
+            responseDocument.appendChild(node);
+        } catch (DOMException e) {
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Nepodarilo sa načítať XML dokument", e);
+        }
+
+        return responseDocument;
+    }
+
+    public static String transformElementToString(Node element) throws XMLValidationException {
+        try {
+            var builderFactory = DocumentBuilderFactory.newInstance();
+            builderFactory.setNamespaceAware(true);
+            builderFactory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+
+            var document = builderFactory.newDocumentBuilder().newDocument();
+            Node node;
+            try {
+                node = document.importNode(element, true);
+            } catch (DOMException e) {
+                node = document.importNode(element.getFirstChild(), true);
+            }
+
+            document.appendChild(node);
+
+            var factory = TransformerFactory.newInstance();
+            var transformer = factory.newTransformer();
+            transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            var writer = new StringWriter();
+            transformer.transform(new DOMSource(document), new StreamResult(writer));
+
+            return writer.toString();
+        } catch (Exception e) {
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Nepodarilo sa načítať XML dokument", e);
+        }
+    }
+
+    public static String transform(DSSDocument documentToDisplay, String transformation) throws TransformationException {
+        try {
+            var parsedDocument = getXmlFromDocument(documentToDisplay);
+            var xmlSource = new DOMSource(parsedDocument);
```

Comment:
mierne sa mi nepozdava (vseobecnem nie tu), ze ci nejako zbytocne vela krat neparsujeme to xml, je to pomerne draha operacia a mozno to bude citit na pouzivani.

Reply:
Tento new DOMSource() sa robí maximálne dvakrát (3x, ak toto pred aj po ife). Jedenkrát je to pri XDCBuilder alebo XDCValidator (nikdy nie oba) a jedenkát tuto pri zobrazovaní. A to sú dve tak oddelené časti, že by bolo zlé to nejako spájať. Jedine, že by sme do SigningJobu alebo do SigningParameters pridali ďalší atribút a to rozparsovaný dokument, čo sa mi nepáči, lebo by sa mohlo stať, že v tomto rozparsovanom by sme (programátorskou chybou v budúcnosti) mali dva rozdielne dokumenty.

Snažil som sa aj znížiť počet parsovaní cez DocumentBuilderFactorz.newDocumentBuilder, tak som to dosť poprehadzoval. A zase tam máme časť, ktorá sa volá pri validácii a pri zobrazovaní, čo sú teda dve rozdielne veci. Tiež sa ešte volá pri vytváraní SigningJobu zo súboru z disku, kedy sa rozhoduje, či XML je XDC. Avšak, to sa robí v SigningJobe a zvyšok v SigningParameters. Toto by sa dalo ešte nejako prekopať a spojiť, ale idk.

A vyhodil som v ServerSigningPsrameters, že SigningParameters sa vytvárali vo validácii aj v get-e. Takže je to lepšie, nšž to bolo.

Reply:
Toto je zrele na taku uvahu o spracovani a ako tam co tecie a kedy co potrebujeme. Mne to na prvu vychadza, ze to mozno ma zmysel rozparsovat okamzite ako sa vytvara signing job a nasledne uz pracovat len s rozparsovanym vsetkym.  Zapisme si niekde, teraz neriesime.

Reply:
Hej, ale podpisovať chceme pôvodný súbor, ak to bol asice alebo xdc a podobne. Dám do issue niekam.

Reply:
To sa nevylucuje, ze si tam nechame aj povodne veci.